### DISCUSSION TITLE
403 error accessing skin page

### DISCUSSION THREAD

**Discussion by Nkawu** [Daveiano/weewx-wdc#136] [Q&A] [answered]
I have a few different skins installed in separate folders off the web root on my on my fully working weewx Docker host. All the skins are working 100% e.g. Belchertown, Weather34, Material. However I cannot get **weewx-wdc** to work.

 For **weewx-wdc** I have in `weewx.conf`:
```
[StdReport]
    [[WdcReport]]
        skin = weewx-wdc
        enable = true
        lang = en
        HTML_ROOT = public_html/weewx-wdc
```
and in `skins/weewx-wdc/skins.conf`:
```
[Extras]
    base_path = /weewx-wdc/
```

When I try to access the skin `http://weewx.local:8011/weewx-wdc/`, I get a "403 Forbidden".

However, `http://weewx.local:8011/weewx-wdc/celestial.html` works 100%. Is a `index.html` file supposed to be generated in `public_html/weewx-wdc`?

Update:
Just noticed the index, week, month & year.html files aren't being generated

**Comment by Nkawu** [comment]
I believe my weewx version may be < 4.9 - confirming

**Comment by Daveiano** [comment]
The 403 is because there is no index.html file inside the weewx-wdc web folder. And that's presumably because you are using weewx < 4.9. Please update to at least WeeWX 4.9 in order to use the skin!

**Reply by Nkawu** [reply]
I just confirmed I'm using weewx 4.10.2. No error messages in the logs, but still no (index|week|month|year).html files being generated

<img width="377" alt="image" src="https://user-images.githubusercontent.com/1527620/227666438-35d1acb4-8520-4f39-87d2-dae36abeb09b.png">

**Reply by Daveiano** [reply]
Could you please provide the log from a complete report run?

Please also set `debug = 1` in weewx.conf, see also here: https://github.com/weewx/weewx/wiki/Help!-Posting-to-weewx-user

Another hint: The skin requires Python 3, are you running Python 2 or 3?

**Reply by Nkawu** [reply]
Think I found it, the error is `type object 'datetime.datetime' has no attribute 'fromisoformat'`, which means python < 3.7

I'm running with an Ubuntu 18.04 LTS base image, which comes with python 3.6.9 and is too much of a hassle to install python 3.7 or 3.8. I'll rebuild with Ubuntu 20.04 LTS base image and test again.

**Comment by Nkawu** [comment, marked as answer]
In addition to WeeWx 4.9+, the skin also requires **Python 3.7+** in order to avoid the `datetime.datetime` error.

Everything works now!

**Reply by Daveiano** [reply]
Good catch, I updated the readme accordingly. Thank you very much!
